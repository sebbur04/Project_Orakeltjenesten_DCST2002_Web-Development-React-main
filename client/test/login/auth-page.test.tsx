// Import neccessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import PageContent from "../../src/login/page";

// Mock services
jest.mock("../../src/service/versionlog-service", () => ({
  getLatestVersion: jest.fn(() =>
    Promise.resolve({
      id: 1,
      pageid: 1,
      content: "Sample content",
      version: 1,
      changelog: "Initial version",
    })
  ),
  createVersion: jest.fn(() => Promise.resolve()),
}));

jest.mock("../../src/service/page-service", () => ({
  getPage: jest.fn(() =>
    Promise.resolve({ id: 1, name: "Test Page", num_views: 10 })
  ),
  getTagsFromPage: jest.fn(() =>
    Promise.resolve([
      { id: 1, name: "Tag1" },
      { id: 2, name: "Tag2" },
    ])
  ),
}));

jest.mock("../../src/service/comment-service", () => ({
  getCommentsFromPage: jest.fn(() =>
    Promise.resolve([
      {
        id: 1,
        date: "2023-10-10T10:00:00Z",
        content: "First comment",
        userid: 1,
        pageid: 1,
      },
    ])
  ),
  createComment: jest.fn(() => Promise.resolve(2)),
  updateComment: jest.fn(() => Promise.resolve()),
  deleteComment: jest.fn(() => Promise.resolve()),
}));

jest.mock("../../src/service/user-service", () => ({
  getAuthenticatedUser: jest.fn(() =>
    Promise.resolve({
      id: 1,
      username: "User1",
      avatar: "avatar.png",
      bio: "Bio",
      permid: 1,
    })
  ),
  getAllUsers: jest.fn(() =>
    Promise.resolve([
      { id: 1, username: "User1", avatar: "avatar.png", bio: "Bio", permid: 1 },
      {
        id: 2,
        username: "User2",
        avatar: "avatar2.png",
        bio: "Bio2",
        permid: 2,
      },
    ])
  ),
}));

jest.mock("../../src/service/permission-service", () => ({
  getPermission: jest.fn(() =>
    Promise.resolve({
      id: 1,
      alterpages: true,
      deletepages: true,
      versions: true,
      allcomments: true,
      tags: true,
      users: true,
    })
  ),
}));

jest.mock("../../src/service/tag-service", () => ({
  getAllTags: jest.fn(() =>
    Promise.resolve([
      { id: 1, name: "Tag1" },
      { id: 2, name: "Tag2" },
    ])
  ),
}));

describe("PageContent Component Tests", () => {
  test("Renders PageContent correctly (snapshot)", async () => {
    const wrapper = shallow(<PageContent match={{ params: { id: 1 } }} />);

    // Wait for asynchronous operations to complete
    // @ts-ignore
    await wrapper.instance().mounted();
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });
});
