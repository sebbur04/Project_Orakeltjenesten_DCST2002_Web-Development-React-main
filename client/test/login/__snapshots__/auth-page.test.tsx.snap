// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent Component Tests Renders PageContent correctly (snapshot) 1`] = `
<div
  className="mainContent"
>
  <Card
    title="Test Page"
  >
    <Row>
      <Column>
        Tagger:
      </Column>
      <div
        className="tags-flexbox"
      >
        <a
          className="blue-link"
          href="/#/tags/1"
          key="1"
        >
          Tag1
        </a>
        <a
          className="blue-link"
          href="/#/tags/2"
          key="2"
        >
          Tag2
        </a>
      </div>
    </Row>
    <div
      className="page-content"
    >
      <Qe>
        Sample content
      </Qe>
    </div>
    <Row>
      <Column
        width={4}
      >
        Antall visninger: 
        10
      </Column>
      <Column
        width={4}
      >
        <a
          className="blue-link"
          href="/#/pages/1/versionlog"
        >
          Versjonslogg
        </a>
      </Column>
      <Column
        width={1}
      >
        <ButtonLight
          onClick={[Function]}
        >
          Rediger
        </ButtonLight>
      </Column>
    </Row>
  </Card>
  <Card
    title="Kommentarer"
  >
    <div
      className="comment-border"
      key="1"
    >
      <Row>
        <Row>
          <Column
            width={12}
          >
            <p>
              <img
                className="comment-image"
                src="avatar.png"
              />
               
              <a
                className="blue-link"
                href="/#/users/1"
              >
                User1
              </a>
            </p>
          </Column>
        </Row>
        <Row>
          <div
            className="small-text-page-comment"
          >
            <Column
              width={12}
            >
              2023-10-10 12:00
            </Column>
          </div>
        </Row>
        <Row>
          <br />
        </Row>
         
        <Row>
          <Column
            width={10}
          >
            <Row>
              <div>
                First comment
              </div>
            </Row>
          </Column>
          <Column
            width={2}
          >
            <div>
              <Column>
                <ButtonLight
                  onClick={[Function]}
                >
                  ✏️
                </ButtonLight>
              </Column>
              <Column>
                <ButtonDanger
                  onClick={[Function]}
                >
                  X
                </ButtonDanger>
              </Column>
            </div>
          </Column>
        </Row>
      </Row>
    </div>
    <Row>
      <div
        className="comment-border"
      >
        <Column
          width={9}
        >
          <FormTextarea
            onChange={[Function]}
            placeholder="Skriv en ny kommentar..."
            rows={4}
            value=""
            width="70vw"
          />
        </Column>
        <Column
          width={1}
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Kommenter
          </ButtonSuccess>
        </Column>
      </div>
    </Row>
  </Card>
</div>
`;
